instance_pool.go

```diff
@@ -0,0 +1,164 @@
+package egoscale
+
+// InstancePoolState reprsents a state of an instance pool
```

Comment:
```suggestion
// InstancePoolState represents the state of an instance pool.
```

Reminder: the convention for Go function comments is to end with a comma (applicable to all of the comments in this PR).